### DISCUSSION TITLE
MLEADER testing

### DISCUSSION THREAD

**Discussion by mozman** [mozman/ezdxf#264] [Ideas]
The generic implementation for loading and exporting tags of the entity MLEADER (MULTILEADER) has been replaced in version 0.15a4 by explicit DXF attributes and embedded data structures.

Because of the complexity of the MLEADER entity I need some help to test this feature, and I assume BricsCAD does not support all possible features very well. So I need test data with MTEXT or BLOCK content with one or more attributes and leaders with (multiple) breaks in all lines and as many overridden MLEADERSTYLE properties as possible.

Please post test files here or send it to me@ezdxf.at and post related errors here.

FYI: Creating new MLEADER entities and rendering is not yet implemented in ezdxf!

Thank you!

**Comment by mbway** [comment]
Hi. My colleague helped to put together a complex MLEADER example with the features that you requested. Hopefully this covers everything? Let me know if you need anything else.

Is MLEADER drawing going to be implemented along with the loading and saving support?

![MLEADER](https://user-images.githubusercontent.com/4923501/99249262-ba438e80-2801-11eb-8968-b23a4cde504f.png)

[MLEADER.dxf.zip](https://github.com/mozman/ezdxf/files/5545999/MLEADER.dxf.zip)

**Comment by mozman** [comment]
Thank you!

Rendering of MLEADER is planned, but only for the 0.16 release. I will release 0.15 when the MLEADER load and export support is stable and lossless.

**Comment by mozman** [comment]
As I expected the BricsCAD support for MLEADER is not perfect:
![image](https://user-images.githubusercontent.com/4230966/99495827-84beb280-2973-11eb-8d6f-4583d3d477c7.png)

Added now proxy graphic support as planned for the future and the results for MLEADER are promising, the color of the "dogleg" below M2 is wrong and the filling flag is not supported yet:

![image](https://user-images.githubusercontent.com/4230966/99495607-2f82a100-2973-11eb-85f6-5b7e49c402e7.png)

But not every CAD application adds proxy graphics to the MLEADER entity, so rendering support for the MLEADER entity is still on the todo list for v0.16.
